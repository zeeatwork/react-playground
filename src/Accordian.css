ul li {
  list-style: none;
}

li {
  border: darkblue thin solid ;
  border-collapse: collapse;
  background-color: #F97068;
}

button {
background-color: #D1D646;
width: 100%;
}