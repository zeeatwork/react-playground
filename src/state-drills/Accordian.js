import React from 'react';
import '../Accordian.css';


export default class Accordian extends React.Component {
  static defaultProps = {sections:[]}

  state = {
    lastClicked: null,
  }
  handleClick = (index) => {
    this.setState({ lastClicked: index })
  }
  render() {
    const sections = this.props.sections;
    const liSections = sections.map((section, index) => {
      return (
        <li key={index}>
          <button onClick={() => this.handleClick(index)}>{section.title}</button>
          {this.state.lastClicked === index && <p>{section.content}</p>}
        </li>
      )
    })
    return (
      <ul>
        {liSections}
      </ul>
    )
  }
}
