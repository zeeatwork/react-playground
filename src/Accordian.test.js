import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import Accordian from './state-drills/Accordian';

describe(`Accordian Component`, () => {
  const data = [ 
    { title: 'Section 1', 
    content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.', }, 
    { title: 'Section 2', 
    content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!', }, 
    { title: 'Section 3', 
    content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?', }, 
  ]
  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordian sections={data}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  })
  
  it ("Component renders no sections as active by default", () => {
    const tree = renderer.create(<Accordian sections={data}/>).toJSON();
    expect (tree).toMatchSnapshot();
  }
  
  )
  it('renders empty given no tabs', () => {
    const wrapper = shallow(<Accordian />)
    toJson(wrapper)
  })
  
  it('renders empty given no tabs', () => { const wrapper = shallow(<Accordian />);
     expect(toJson(wrapper)).toMatchSnapshot() })
  
     it('closes the first tab and opens any clicked tab', () => {
      const wrapper = shallow(<Accordian sections={data} />)
      wrapper.find('button').at(1).simulate('click')
      expect(toJson(wrapper)).toMatchSnapshot()
    })
})
