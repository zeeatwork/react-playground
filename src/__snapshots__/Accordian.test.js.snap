// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordian Component Component renders no sections as active by default 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Component renders no sections as active by default 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`renders empty given no sections 1`] = `
<ul></ul>
`;
